### PR TITLE
feat: define typescript components interface

### PR DESCRIPTION
**Pull request by TahimiLeonBravo**
#942 

## Changes proposed in this PR:
- define typescript components interface

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/Badge/index.d.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
-export interface BadgeProps {
+export interface BadgeProps extends BaseProps {
```
Comment (LeandroTorresSicilia) [inline src/components/Badge/index.d.ts]: BadgeProps?

File: src/components/Badge/index.d.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
     label?: string | JSX.ElementChildrenAttribute;
     title?: string;
     variant?: 'default' | 'inverse' | 'lightest' | 'outline-brand' | 'brand';
-    className?: string;
-    style?: CSS.style;
 }
 
 export default function(props: BadgeProps): JSX.Element | null;
```
Comment (LeandroTorresSicilia) [inline src/components/Badge/index.d.ts]: same ^^

**Comment by stephen-zhao** [comment]
I think `export default` should be avoided. See: https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html

What do you think?

File: src/components/Button/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from 'react';
+import { BaseProps } from './../types';
+
+export interface ButtonProps extends BaseProps {
+    label?: string | JSX.ElementChildrenAttribute;
+    variant?:
+        | 'base'
```
Comment (stephen-zhao) [inline src/components/Button/index.d.ts]: that's cool, didn't know leading operators were a thing 😁

File: src/components/ButtonIcon/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    disabled?: boolean;
+    form?: string;
+    icon: JSX.ElementChildrenAttribute;
+    id?: string;
+    // onBlur?: ;
+    // onClick?: ;
+    // onFocus?: ;
+    // onKeyDown?: ;
+    shaded?: boolean;
+    size?: 'xx-small' | 'x-small' | 'small' | 'medium' | 'large';
```
Comment (stephen-zhao) [inline src/components/ButtonIcon/index.d.ts]: Would there be any benefit of type-aliasing these union types to share across files? I noticed there is a lot of repeat. I can see maintainability issues if we need to rename or add a new size in the future...

Type aliases are resolved before showing up in IDEs like VS Code, so they won't affect clients' workflow.

**Comment by reiniergs** [comment]
@stephen-zhao agree with you about `export default`, but at this point, the library expects the components exported as default in `index.js` files and the declaration files need to mock the component implementation, basically there is not much we can do :)

**Pull request merged**